shell/platform/linux/fl_json_message_codec_test.cc

```diff
@@ -0,0 +1,769 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_message_codec.h"
+#include "gtest/gtest.h"
+
+#include <math.h>
+
+// TODO(robert-ancell): Test decoding integers that can't fit into 64 bits
```

Comment:
Per other discussion I don't think this should be possible to reach.

Reply:
I've added tests for the large integer cases.